cocos/rendering/custom/layout-graph.ts

```diff
@@ -1014,6 +1308,171 @@ export class LayoutGraphObjectPool {
     private readonly lgd: RecyclePool<LayoutGraphData> = createPool(LayoutGraphData);
 }
 
+export function saveDescriptor (a: OutputArchive, v: Descriptor): void {
+    a.n(v.type);
+    a.n(v.count);
+}
+
+export function loadDescriptor (a: InputArchive, v: Descriptor): void {
+    v.type = a.n();
+    v.count = a.n();
+}
+
+export function saveDescriptorBlock (a: OutputArchive, v: DescriptorBlock): void {
+    a.n(v.descriptors.size); // Map<string, Descriptor>
+    for (const [k1, v1] of v.descriptors) {
+        a.s(k1);
+        saveDescriptor(a, v1);
+    }
+    a.n(v.uniformBlocks.size); // Map<string, UniformBlock>
+    for (const [k1, v1] of v.uniformBlocks) {
+        a.s(k1);
+        saveUniformBlock(a, v1);
+    }
+    a.n(v.capacity);
+    a.n(v.count);
+}
+
+export function loadDescriptorBlock (a: InputArchive, v: DescriptorBlock): void {
+    let sz = 0;
+    sz = a.n(); // Map<string, Descriptor>
+    for (let i1 = 0; i1 !== sz; ++i1) {
+        const k1 = a.s();
+        const v1 = new Descriptor();
+        loadDescriptor(a, v1);
+        v.descriptors.set(k1, v1);
+    }
+    sz = a.n(); // Map<string, UniformBlock>
+    for (let i1 = 0; i1 !== sz; ++i1) {
+        const k1 = a.s();
+        const v1 = new UniformBlock();
+        loadUniformBlock(a, v1);
+        v.uniformBlocks.set(k1, v1);
+    }
+    v.capacity = a.n();
+    v.count = a.n();
+}
+
+export function saveDescriptorBlockFlattened (a: OutputArchive, v: DescriptorBlockFlattened): void {
+    a.n(v.descriptorNames.length); // string[]
+    for (const v1 of v.descriptorNames) {
+        a.s(v1);
+    }
+    a.n(v.uniformBlockNames.length); // string[]
+    for (const v1 of v.uniformBlockNames) {
+        a.s(v1);
+    }
+    a.n(v.descriptors.length); // Descriptor[]
+    for (const v1 of v.descriptors) {
+        saveDescriptor(a, v1);
+    }
+    a.n(v.uniformBlocks.length); // UniformBlock[]
+    for (const v1 of v.uniformBlocks) {
+        saveUniformBlock(a, v1);
+    }
+    a.n(v.capacity);
+    a.n(v.count);
+}
+
+export function loadDescriptorBlockFlattened (a: InputArchive, v: DescriptorBlockFlattened): void {
```

Comment:
The parameters `a` & `v` are not detailed. 
Parameter names could be mangled/minified by terser.

Reply:
I will rename these values in another pr.